polkadot/api/src/lib.rs

```diff
@@ -93,40 +94,64 @@ pub trait BlockBuilder: Sized {
 	fn bake(self) -> Block;
 }
 
+/// A checked block identifier.
+pub trait CheckedBlockId: Clone {
+	/// Yield the underlying block ID.
+	fn block_id(&self) -> &BlockId;
```

Comment:
worth making this implement `AsRef`?

Reply:
potentially, but it's more overhead to implement the trait for any types